### PR TITLE
Fixing nist.Int.Clone()

### PR DESCRIPTION
**Pull request by nikkolasg**
Value was copied but value is a big.Int which contains the real value which is a slice (i.e. pointer).

### CODE REVIEW

File: nist/int.go [hunk 1]
```diff
@@ -140,7 +140,9 @@ func (i *Int) Set(a abstract.Scalar) abstract.Scalar {
 }
 
 func (i *Int) Clone() abstract.Scalar {
-	return &Int{V: i.V, M: i.M, BO: i.BO}
+	v := new(big.Int).Set(&i.V)
+	m := new(big.Int).Set(i.M)
+	return &Int{V: *v, M: m, BO: i.BO}
```
Comment (ineiti) [inline nist/int.go]: `return new(Int).Init(&i.V, i.M)`

Comment (nikkolasg) [inline nist/int.go]: But then the ByteOrder may not be correct.

File: nist/int_test.go [hunk 2]
```diff
@@ -63,3 +63,22 @@ func TestIntEndianBytes(t *testing.T) {
 	assert.Equal(t, 2, i.MarshalSize())
 	assert.NotPanics(t, func() { i.LittleEndian(2, 2) })
 }
+
+func TestIntClone(t *testing.T) {
+	modulo, err := hex.DecodeString("1000")
+	moduloI := new(big.Int).SetBytes(modulo)
```
Comment (ineiti) [inline nist/int_test.go]: Does
`moduloI := new(big.Int).SetBytes(0x1000)`
not work?

File: nist/int_test.go [hunk 3]
```diff
@@ -63,3 +63,22 @@ func TestIntEndianBytes(t *testing.T) {
...
+	assert.Nil(t, err)
+	v, err := hex.DecodeString("10")
+	assert.Nil(t, err)
+
+	base := new(Int).InitBytes(v, moduloI)
```
Comment (ineiti) [inline nist/int_test.go]: Same thing - `0x10` instead of `hex.DecodeString`?

File: nist/int.go (hunk 1 above)
Comment (ineiti) [inline nist/int.go]: yes, so we have to add that, too... I saw it does not correctly marshal/unmarshal 'my' way.

Comment (ineiti) [inline nist/int.go]: So the complete is:
```
	ni := new(Int).Init(&i.V, i.M)
	ni.BO = i.BO
	return ni
```

So either we take your solution, or we re-write `Int.Init` to take `BO` as a parameter, which might makes some sense, given that the `V` -argument to `Init` could be in the wrong endianness. But I don't know if `BO` is really relevant in `i.V.Set(V).Mod(&i.V, M)` - probably not.

Your call, as Ismail is not here to vote ;)

**Comment by nikkolasg** [comment]
Can we go back to the simpler version without changing nist.Int  and let #90 decide on that matter ?

**Comment by ineiti** [comment]
Replaced by #87

**Pull request closed without merging**